From adcea644f4158f7bd7c8a61b3af3ca795b8c42bc Mon Sep 17 00:00:00 2001
From: Robert Kanter <rkanter@cloudera.com>
Date: Wed, 17 Sep 2014 11:43:24 -0700
Subject: [PATCH 150/596] MR1: CLOUDERA-BUILD. A new simple fix to allow
 native libs to build on Ubuntu 14.04 (CDH-20987)

(cherry picked from commit 09bee171f7d7bb9c83bd214f5eece957eb07eee4)
---
 .../src/c++/task-controller/impl/main.c            |   16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/c++/task-controller/impl/main.c b/hadoop-mapreduce1-project/src/c++/task-controller/impl/main.c
index 770fdf0..75b1b25 100644
--- a/hadoop-mapreduce1-project/src/c++/task-controller/impl/main.c
+++ b/hadoop-mapreduce1-project/src/c++/task-controller/impl/main.c
@@ -143,18 +143,14 @@ int main(int argc, char **argv) {
   }
   set_tasktracker_uid(getuid(), group_info->gr_gid);
   // if we are running from a setuid executable, make the real uid root
-  // CDH-20987: To allow compiling on Ubuntu 14.04, we need to add this dummy
-  // variable
-  int dummy1 = setuid(0);
-  if (dummy1 != 0) {
-    dummy1 = 0;
+  int result = setuid(0);
+  if (result != 0) {
+    return SETUID_OPER_FAILED;
   }
   // set the real and effective group id to the task tracker group
-  // CDH-20987: To allow compiling on Ubuntu 14.04, we need to add this dummy
-  // variable
-  int dummy2 = setgid(group_info->gr_gid);
-  if (dummy2 != 0) {
-    dummy2 = 0;
+  result = setgid(group_info->gr_gid);
+  if (result != 0) {
+    return SETUID_OPER_FAILED;
   }
 
   if (check_taskcontroller_permissions(executable_file) != 0) {
-- 
1.7.9.5

